<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.33">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Martini 3 protein models - a practical introduction to different structure bias models and their comparison – Martini Force Field Initiative</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../../">
<link href="../../../../images/favicon.png" rel="icon" type="image/png">
<script src="../../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../../site_libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../../site_libs/bootstrap/bootstrap-69fe911abf326d3c3f52f7dd89351358.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../../../site_libs/quarto-contrib/iconify-2.1.0/iconify-icon.min.js"></script>
<link href="../../../../site_libs/quarto-contrib/fontawesome6-1.2.0/all.min.css" rel="stylesheet">
<link href="../../../../site_libs/quarto-contrib/fontawesome6-1.2.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>
    document.addEventListener('DOMContentLoaded', (event) => {
      const quotes = [
          "<i>Shaken, not stirred.</i> - James Bond",
          "<i>He was white and shaken, like a dry martini.</i> - P. G. Wodehouse",      
          "<i>I think people think we're all sipping martinis by the pool.</i> - Maria Menounos",
          "<i>Why don't you get out of that wet coat and into a dry martini?</i> - Robert Benchley",
          "<i>Really, can anyone drink several martinis at lunch?</i> - Christine Baranski",
          "<i>Zen martini: A martini with no vermouth at all. And no gin, either.</i> - P.J. O'Rourke",
          "<i>I never go jogging, it makes me spill my martini.</i> - George Burns",
          "<i>Martinis are the only American invention as perfect as a sonnet.</i> - H. L. Mencken",
          "<i>He knows just how I like my martini: full of alcohol.</i> - Homer Simpson",
          "<i>Happiness is: finding olives in your martini when you're hungry.</i> - Johnny Carson",
          "<i>If it wasn't for the olives in his martinis he'd starve to death.</i> - Milton Berle",
        ];
      const randomIndex = Math.floor(Math.random() * quotes.length);
      const randomQuote = quotes[randomIndex];

      const footerLeft = document.querySelector('.footer-left');
      if (footerLeft) {
          footerLeft.innerHTML = randomQuote;
      }
    });
</script>


<link rel="stylesheet" href="../../../../styles.css">
<meta property="og:title" content="Martini 3 protein models - a practical introduction to different structure bias models and their comparison – Martini Force Field Initiative">
<meta property="og:description" content="">
<meta property="og:image" content="https://cgmartini.nl/docs/tutorials/Martini3/ProteinsI/Fig1.png">
<meta property="og:site_name" content="Martini Force Field Initiative">
<meta property="og:image:height" content="5520">
<meta property="og:image:width" content="6519">
<meta name="twitter:title" content="Martini 3 protein models - a practical introduction to different structure bias models and their comparison – Martini Force Field Initiative">
<meta name="twitter:description" content="">
<meta name="twitter:image" content="https://cgmartini.nl/docs/tutorials/Martini3/ProteinsI/Fig1.png">
<meta name="twitter:image-height" content="5520">
<meta name="twitter:image-width" content="6519">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../../images/martini.gif" alt="martini logo" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../../index.html"> <i class="bi bi-house-fill" role="img">
</i> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../../../docs/tutorials/index.html" aria-current="page"> <i class="bi bi-mortarboard-fill" role="img">
</i> 
<span class="menu-text">Tutorials</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/downloads/index.html"> <i class="bi bi-download" role="img">
</i> 
<span class="menu-text">Downloads</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/publications/index.html"> <i class="bi bi-newspaper" role="img">
</i> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/announcements/index.html"> 
<span class="menu-text"><i class="fa-solid fa-bullhorn" aria-label="bullhorn"></i> Announcements</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/contact/index.html"> 
<span class="menu-text"><i class="fa-regular fa-address-book" aria-label="address-book"></i> Contact</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-fa-circle-info--help" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text"><i class="fa-solid fa-circle-info" aria-label="circle-info"></i> Help</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-fa-circle-info--help">    
        <li>
    <a class="dropdown-item" href="https://github.com/orgs/Martini-Force-Field-Initiative/discussions"><i class="bi bi-chat-right-text" role="img">
</i> 
 <span class="dropdown-text">Discussion Board</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/orgs/Martini-Force-Field-Initiative/discussions"><i class="bi bi-bug" role="img">
</i> 
 <span class="dropdown-text">Report an Issue</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../../docs/faq/index.html"><i class="bi bi-question-circle" role="img">
</i> 
 <span class="dropdown-text">FAQs</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://x.com/cg_martini" title="Martini Twitter" class="quarto-navigation-tool px-1" aria-label="Martini Twitter"><i class="bi bi-twitter"></i></a>
    <a href="https://github.com/Martini-Force-Field-Initiative" title="Martini GitHub" class="quarto-navigation-tool px-1" aria-label="Martini GitHub"><i class="bi bi-github"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Martini 3 protein models - a practical introduction to different structure bias models and their comparison</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><b>Summary</b></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/Martini3/lectures.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">🎓 Martini Lectures</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/Martini3/tutorials.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">🛠️ Hands-on Tutorials</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/Legacy/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">🕸️ Legacy Material</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#table-of-contents" id="toc-table-of-contents" class="nav-link active" data-scroll-target="#table-of-contents">Table of contents</a></li>
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#Tutorials" id="toc-Tutorials" class="nav-link" data-scroll-target="#Tutorials">Tutorial Sections</a>
  <ul class="collapse">
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Martini 3 protein models - a practical introduction to different structure bias models and their comparison</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<hr>
<p>If you would like to refer to information of this tutorial, please cite <a href="https://www.biorxiv.org/content/10.1101/2025.03.17.643608v2">T. Duve, L. Wang, L. Borges-Araújo, S. J. Marrink, P. C. T. Souza, S. Thallmair, Martini 3 Protein Models - A Practical Introduction to Different Structure Bias Models and their Comparison, bioRxiv (2025), doi: 10.1101/2025.03.17.643608</a>.</p>
<p>In case of issues, please contact <a href="mailto:%20duve@fias.uni-frankfurt.de">duve@fias.uni-frankfurt.de</a>, <a href="mailto:%20luis.borges@ens-lyon.fr">luis.borges@ens-lyon.fr</a>, or <a href="mailto:%20thallmair@fias.uni-frankfurt.de">thallmair@fias.uni-frankfurt.de</a>.</p>
<hr>
<section id="table-of-contents" class="level3">
<h3 class="anchored" data-anchor-id="table-of-contents">Table of contents</h3>
<ul>
<li><a href="#introduction"><code>Introduction</code></a></li>
<li><a href="#Tutorials"><code>Tutorial Sections</code></a>
<ul>
<li><p><a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut1.html"><code>Tutorial I.I: Setup of Martini 3 Protein Structure Models</code></a></p></li>
<li><p><a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut2.html"><code>Tutorial I.II: Metrics to Compare Protein Structure Models</code></a></p></li>
<li><p><a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut3.html"><code>Tutorial I.III: Simulations of IDRs in Martini 3</code></a></p></li>
<li><p><a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut4.html"><code>Tutorial I.IV: Additional notes, limitations &amp; conclusion</code></a></p></li>
</ul></li>
<li><a href="#references"><code>References</code></a></li>
</ul>
</section>
<section id="introduction" class="level3">
<h3 class="anchored" data-anchor-id="introduction">Introduction</h3>
<hr>
<!--- Commented to keep intro shorter. Users should already know this bit from earlier tutorials + lectures. --->
<!--- A detailed biophysical characterization of proteins including their structural dynamics is crucial for a broad variety of research fields ranging from molecular biology and drug development to enzyme design and environmentally friendly chemistry. In recent years, modelling techniques became more and more routine tools, offering a complementary approach to experimental techniques for studying protein biophysics. Atomistic and coarse-grained (CG) molecular dynamics (MD) simulations are among the key techniques applied in the field [1, 2]. In this chapter, we will provide a hands-on introduction to the protein models available in the fully re-parametrized general-purpose Martini 3 CG force field [3]. We assume that the reader has hands-on experience with the program package GROMACS [4], for which excellent tutorials are available[5]. --->
<!--- Martini 3 combines top-down and bottom-up coarse-graining with a general mapping of 4-to-1 non-hydrogen atoms to one regular (R) bead [3]. Small (S, 3-to-1 mapping) and tiny (T, 2-to-1 mapping) are also available. The non-bonded interactions are parametrized based on experimental thermodynamic data, while the bonded interactions are parametrized using atomistic[6–8] or quantum mechanical reference simulations [7, 9]. We refer the reader for more details to some recent reviews [10, 11]. --->
<p>The Martini 3 protein model comprises two layers, which are strictly separated from each other: The first layer contains the mapping and chosen chemical bead types as well as the bonded terms; the second layer is the structure bias model. While the first layer corresponds to the standard definition of any molecule in the Martini universe, the second layer is more specific for proteins: To maintain the secondary, tertiary and quaternary structure of proteins, the directionality of interactions, in particular hydrogen bonds, are crucial. Due to the spherical potentials used in Martini 3 and many other CG force fields, this directionality is lost. Therefore, a structure bias model is required to stabilize protein structures. The strict separation between the two layers in Martini 3 enables independent development on both layers and is in contrast to the Martini 2 model[12], where mappings, bead types, and bonded terms were specific to each of the protein models. The first layer of the current Martini 3 protein model is mostly inherited from the Martini 2 model with moderate updates concerning the new mapping guidelines, bead types, and sizes. The Martini 3 mappings and bead types of the proteinogenic amino acids are depicted in Figure 1. A key difference to the previous model is that the bead type of the backbone (BB) bead does not depend on the secondary structure anymore but is represented by a P2 bead. Exceptions are the special cases glycine (SP1), alanine (SP2), valine (SP2), proline (SP2a), and terminal beads (Q5 for charged and P6 for neutral termini) [13].</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Fig1.png" class="img-fluid figure-img"></p>
<figcaption>Figure 1: Mapping and chemical bead types of amino acids in the Martini 3 protein model. The main classes of chemical bead types are indicated by color: P (polar, in red), N (intermediately polar, in blue), C (nonpolar, in gray), and Q (charged, in lime). Different bead sizes are indicated by the radius of the bead, with regular (no symbol), small (S), and tiny (T) beads.</figcaption>
</figure>
</div>
<p>Here, we will focus on the setup, fine-tuning options, and evaluation of the second layer of the Martini 3 protein model – the structure bias model – as well as on dedicated options for intrinsically disordered regions (IDRs). Overall, the Martini protein models require specific bonded terms to model the secondary structure [10, 12], which is in most cases combined with side-chain dihedral corrections [14]<strong>.</strong> In addition, there are three main options for structural bias models in layer 2 available, typically used to stabilize tertiary and quaternary structures, namely Elastic Network (EN) [3, 15, 16], GōMartini [13], and OLIVES [17] (Figure 2). Whereas the ad-hoc EN approach provides the most robust and straightforward way of stabilizing a protein structure, and has traditionally been used in Martini, the bioinformatics-based GōMartini approach is nowadays the method of choice as it has proven to be a more versatile method, striking a balance between protein stability and flexibility. Most recently, OLIVES was introduced as a physics-based variant of GōMartini, with the particular prospect of biasing quaternary structures as well.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Fig2.png" class="img-fluid figure-img"></p>
<figcaption>Figure 2: Schematic visualization of the two-layered Martini 3 protein model.</figcaption>
</figure>
</div>
<p>In their simplest form, each of these three structure bias models relies on an atomistic reference structure which can be obtained from experiments or prediction software. The EN model uses harmonic potentials between the BB beads within a cutoff distance to maintain the protein structure. In contrast, both the GōMartini and OLIVES models employ Lennard-Jones potentials in a Gō-type manner, allowing for contact dissociation. In the GōMartini model, the potentials are defined from a contact map based on native contacts, evaluated through overlap and restricted contacts of structural units [18, 19], combined with a distance cutoff. The OLIVES model, on the other hand, defines its bias by evaluating the hydrogen bonding interactions between BB as well as side chain (SC) beads [17]. For small structured systems, such as single α-helices, the waiver of a structural bias model can be an option as well.</p>
<!--- The first section, Section I.I, will show you how to build your first Martini 3 protein model. --->
</section>
<section id="Tutorials" class="level2">
<h2 class="anchored" data-anchor-id="Tutorials">Tutorial Sections</h2>
<p>The tutorial is structured as follows:</p>
<ul>
<li><p><a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut1.html"><strong><code>Tutorial I.I: Setup of Martini 3 Protein Structure Models</code></strong></a></p>
<ul>
<li>Tutorial I.I provides details on how to set up a Martini 3 protein model, taking a protein kinase as our workhorse, and using either EN, GōMartini, or OLIVES, together with secondary structure-specific bonded terms and side-chain corrections.</li>
</ul></li>
<li><p><a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut2.html"><strong><code>Tutorial I.II: Metrics to Compare Protein Structure Models</code></strong></a></p>
<ul>
<li>The evaluation of the performance of the model in comparison to atomistic reference simulations is discussed in Section Tutorial I.II.</li>
</ul></li>
<li><p><a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut3.html"><strong><code>Tutorial I.III: Simulations of IDRs in Martini 3</code></strong></a></p>
<ul>
<li>Tutorial I.III gives an example for the addition of an IDR to the structured domain of the kinase.</li>
</ul></li>
<li><p><a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut4.html"><strong><code>Tutorial I.IV: Additional notes, limitations &amp; conclusion</code></strong></a></p>
<ul>
<li>Additional notes and limitations are discussed in Tutorial I.IV.1, before we conclude in Tutorial I.IV.2.</li>
</ul></li>
</ul>
<!--- Related Martini-based tutorial chapters are available for constructing PEGylated proteins [20], embedding membrane proteins [21], and for protein-ligand binding [7].  --->
<hr>
<section id="references" class="level3">
<h3 class="anchored" data-anchor-id="references">References</h3>
<ol type="1">
<li>Borges-Araújo L, Patmanidis I, Singh AP, et al (2023) Pragmatic Coarse-Graining of Proteins: Models and Applications. J Chem Theory Comput 19:7112–7135. https://doi.org/10.1021/acs.jctc.3c00733</li>
<li>Samuel Russell PP, Alaeen S, Pogorelov TV (2023) In-Cell Dynamics: The Next Focus of All-Atom Simulations. J Phys Chem B 127:9863–9872. https://doi.org/10.1021/acs.jpcb.3c05166</li>
<li>Souza PCT, Alessandri R, Barnoud J, et al (2021) Martini 3: a general purpose force field for coarse-grained molecular dynamics. Nat Methods 18:382–388. https://doi.org/10.1038/s41592-021-01098-3</li>
<li>Abraham MJ, Murtola T, Schulz R, et al (2015) GROMACS: High performance molecular simulations through multi-level parallelism from laptops to supercomputers. SoftwareX 1–2:19–25. https://doi.org/10.1016/j.softx.2015.06.001</li>
<li>Lemkul JA (2024) Introductory Tutorials for Simulating Protein Dynamics with GROMACS. J Phys Chem B 128:9418–9435. https://doi.org/10.1021/acs.jpcb.4c04901</li>
<li>Alessandri R, Barnoud J, Gertsen AS, et al (2022) Martini 3 Coarse-Grained Force Field: Small Molecules. Adv Theory Simul 5:2100391. https://doi.org/10.1002/adts.202100391</li>
<li>Alessandri R, Thallmair S, Herrero CG, et al (2023) A Practical Introduction to Martini 3 and its Application to Protein-Ligand Binding Simulations. https://doi.org/10.1063/9780735425279_001</li>
<li>Gil Herrero C, Thallmair S (2024) G-Protein-Coupled Receptor Surface Creates a Favorable Pathway for Membrane Permeation of Drug Molecules. J Phys Chem Lett 15:12643–12651. https://doi.org/10.1021/acs.jpclett.4c02875</li>
<li>Pereira GP, Alessandri R, Domínguez M, et al (2024) Bartender: Martini 3 Bonded Terms via Quantum Mechanics-Based Molecular Dynamics. J Chem Theory Comput 20:5763–5773. https://doi.org/10.1021/acs.jctc.4c00275</li>
<li>Borges-Araújo L, Pereira GP, Valério M, Souza PCT (2024) Assessing the Martini 3 protein model: A review of its path and potential. Biochim Biophys Acta BBA - Proteins Proteomics 1872:141014. https://doi.org/10.1016/j.bbapap.2024.141014</li>
<li>Marrink SJ, Monticelli L, Melo MN, et al (2023) Two decades of Martini: Better beads, broader scope. WIREs Comput Mol Sci 13:e1620. https://doi.org/10.1002/wcms.1620</li>
<li>Monticelli L, Kandasamy SK, Periole X, et al (2008) The MARTINI Coarse-Grained Force Field: Extension to Proteins. J Chem Theory Comput 4:819–834. https://doi.org/10.1021/ct700324x</li>
<li>Souza PCT, Araujo LPB, Brasnett C, et al (2024) GōMartini 3: From large conformational changes in proteins to environmental bias corrections. 2024.04.15.589479</li>
<li>Herzog FA, Braun L, Schoen I, Vogel V (2016) Improved Side Chain Dynamics in MARTINI Simulations of Protein–Lipid Interfaces. J Chem Theory Comput 12:2446–2458. https://doi.org/10.1021/acs.jctc.6b00122</li>
<li>Kroon PC, Grunewald F, Barnoud J, et al (2024) Martinize2 and Vermouth: Unified Framework for Topology Generation. eLife 12:. https://doi.org/10.7554/eLife.90627.2</li>
<li>Periole X, Cavalli M, Marrink S-J, Ceruso MA (2009) Combining an Elastic Network With a Coarse-Grained Molecular Force Field: Structure, Dynamics, and Intermolecular Recognition. J Chem Theory Comput 5:2531–2543. https://doi.org/10.1021/ct9002114</li>
<li>Pedersen KB, Borges-Araújo L, Stange AD, et al (2024) OLIVES: A Go̅-like Model for Stabilizing Protein Structure via Hydrogen Bonding Native Contacts in the Martini 3 Coarse-Grained Force Field. J Chem Theory Comput 20:8049–8070. https://doi.org/10.1021/acs.jctc.4c00553</li>
<li>Poma AB, Cieplak M, Theodorakis PE (2017) Combining the MARTINI and Structure-Based Coarse-Grained Approaches for the Molecular Dynamics Studies of Conformational Transitions in Proteins. J Chem Theory Comput 13:1366–1374. https://doi.org/10.1021/acs.jctc.6b00986</li>
<li>Wołek K, Gómez-Sicilia À, Cieplak M (2015) Determination of contact maps in proteins: A combination of structural and chemical approaches. J Chem Phys 143:243105. https://doi.org/10.1063/1.4929599</li>
<li>Grünewald F, Kroon PC, Souza PCT, Marrink SJ (2021) Protocol for Simulations of PEGylated Proteins with Martini 3. In: Chen YW, Yiu C-PB (eds) Structural Genomics: General Applications. Springer US, New York, NY, pp 315–335</li>
<li>Ozturk TN, König M, Carpenter TS, et al (2024) Chapter Seven - Building complex membranes with Martini 3. In: Deserno M, Baumgart T (eds) Methods in Enzymology. Academic Press, pp 237–285</li>
<li>Zeringo NA, Murphy L, McCloskey EA, et al (2013) A monoclinic crystal form of casein kinase 1 δ. Acta Crystallograph Sect F Struct Biol Cryst Commun 69:1077–1083. https://doi.org/10.1107/S1744309113023403</li>
<li>Waterhouse A, Bertoni M, Bienert S, et al (2018) SWISS-MODEL: homology modelling of protein structures and complexes. Nucleic Acids Res 46:W296–W303. https://doi.org/10.1093/nar/gky427</li>
<li>Abramson J, Adler J, Dunger J, et al (2024) Accurate structure prediction of biomolecular interactions with AlphaFold 3. Nature 1–3. https://doi.org/10.1038/s41586-024-07487-w</li>
<li>Anandakrishnan R, Aguilar B, Onufriev AV (2012) H++ 3.0: automating pK prediction and the preparation of biomolecular structures for atomistic molecular modeling and simulations. Nucleic Acids Res 40:W537–W541. https://doi.org/10.1093/nar/gks375</li>
<li>Olsson MHM, Søndergaard CR, Rostkowski M, Jensen JH (2011) PROPKA3: Consistent Treatment of Internal and Surface Residues in Empirical pKa Predictions. J Chem Theory Comput 7:525–537. https://doi.org/10.1021/ct100578z</li>
<li>Lee J, Cheng X, Jo S, et al (2016) CHARMM-GUI Input Generator for NAMD, Gromacs, Amber, Openmm, and CHARMM/OpenMM Simulations using the CHARMM36 Additive Force Field. Biophys J 110:641a. https://doi.org/10.1016/j.bpj.2015.11.3431</li>
<li>Jo S, Kim T, Iyer VG, Im W (2008) CHARMM‐GUI: A web‐based graphical user interface for CHARMM. J Comput Chem 29:1859–1865. https://doi.org/10.1002/jcc.20945</li>
<li>Kabsch W, Sander C (1983) Dictionary of protein secondary structure: Pattern recognition of hydrogen-bonded and geometrical features. Biopolymers 22:2577–2637. https://doi.org/10.1002/bip.360221211</li>
<li>McGibbon RT, Beauchamp KA, Harrigan MP, et al (2015) MDTraj: A Modern Open Library for the Analysis of Molecular Dynamics Trajectories. Biophys J 109:1528–1532. https://doi.org/10.1016/j.bpj.2015.08.015</li>
<li>Alessandri R, Souza PCT, Thallmair S, et al (2019) Pitfalls of the Martini Model. J Chem Theory Comput 15:5448–5460. https://doi.org/10.1021/acs.jctc.9b00473</li>
<li>Wassenaar TA, Ingólfsson HI, Böckmann RA, et al (2015) Computational Lipidomics with insane : A Versatile Tool for Generating Custom Membranes for Molecular Simulations. J Chem Theory Comput 11:2144–2155. https://doi.org/10.1021/acs.jctc.5b00209</li>
<li>Andreasen MD, Souza PCT, Schiøtt B, Zuzic L (2024) Creating Coarse-Grained Systems with COBY: Towards Higher Accuracy in Membrane Complexity. 2024.07.23.604601</li>
<li>Marrink S, Peter Tieleman D (2013) Perspective on the Martini model. Chem Soc Rev 42:6801–6822. https://doi.org/10.1039/C3CS60093A</li>
<li>Souza PCT, Thallmair S, Marrink SJ, Mera-Adasme R (2019) An Allosteric Pathway in Copper, Zinc Superoxide Dismutase Unravels the Molecular Mechanism of the G93A Amyotrophic Lateral Sclerosis-Linked Mutation. J Phys Chem Lett 10:7740–7744. https://doi.org/10.1021/acs.jpclett.9b02868</li>
<li>Thomasen FE, Pesce F, Roesgaard MA, et al (2022) Improving Martini 3 for Disordered and Multidomain Proteins. J Chem Theory Comput 18:2033–2041. https://doi.org/10.1021/acs.jctc.1c01042</li>
<li>Larsen AH, Wang Y, Bottaro S, et al (2020) Combining molecular dynamics simulations with small-angle X-ray and neutron scattering data to study multi-domain proteins in solution. PLOS Comput Biol 16:e1007870. https://doi.org/10.1371/journal.pcbi.1007870</li>
<li>Wang L, Brasnett C, Borges-Araújo L, et al (2024) Martini3-IDP: Improved Martini 3 Force Field for Disordered Proteins. submitted</li>
<li>Grünewald F, Alessandri R, Kroon PC, et al (2022) Polyply; a python suite for facilitating simulations of macromolecules and nanomaterials. Nat Commun 13:68. https://doi.org/10.1038/s41467-021-27627-4</li>
<li>Hilpert C, Beranger L, Souza PCT, et al (2023) Facilitating CG Simulations with MAD: The MArtini Database Server. J Chem Inf Model 63:702–710. https://doi.org/10.1021/acs.jcim.2c01375</li>
<li>Yang S, Song C (2024) Multiple-Basin Gō-Martini for Investigating Conformational Transitions and Environmental Interactions of Proteins</li>
<li>Eastman P, Galvelis R, Peláez RP, et al (2024) OpenMM 8: Molecular Dynamics Simulation with Machine Learning Potentials. J Phys Chem B 128:109–116. https://doi.org/10.1021/acs.jpcb.3c06662</li>
<li>Yang S, Song C (2024) Switching Go̅-Martini for Investigating Protein Conformational Transitions and Associated Protein–Lipid Interactions. J Chem Theory Comput 20:2618–2629. https://doi.org/10.1021/acs.jctc.3c01222</li>
<li>Korshunova K, Kiuru J, Liekkinen J, et al (2024) Martini 3 OliGo̅mers: A Scalable Approach for Multimers and Fibrils in GROMACS. J Chem Theory Comput 20:7635–7645. https://doi.org/10.1021/acs.jctc.4c00677</li>
<li>Cofas-Vargas LF, Moreira RA, Poblete S, et al (2024) The Go-Martini Approach: Revisiting the Concept of Contact Maps and the Modelling of Protein Complexes. Acta Phys Pol ISSN 1898-794X 145:S9–S9. https://doi.org/10.12693/APhysPolA.145.S9</li>
<li>Gomes DEB, Melo MCR, Gomes PSFC, Bernardi RC (2022) Bridging the gap between in vitro and in silico single-molecule force spectroscopy. 2022.07.14.500151</li>
<li>Liu Z, Moreira RA, Dujmović A, et al (2022) Mapping Mechanostable Pulling Geometries of a Therapeutic Anticalin/CTLA-4 Protein Complex. Nano Lett 22:179–187. https://doi.org/10.1021/acs.nanolett.1c03584</li>
<li>Humphrey W, Dalke A, Schulten K (1996) VMD: Visual molecular dynamics. J Mol Graph 14:33–38. https://doi.org/10.1016/0263-7855(96)00018-5</li>
<li>Sami S, Grünewald F, Souza PCT, Marrink SJ (2023) A Guide to Titratable Martini Simulations. https://doi.org/10.1063/9780735425279_004</li>
<li>Grünewald F, Souza PCT, Abdizadeh H, et al (2020) Titratable Martini model for constant pH simulations. J Chem Phys 153:024118. https://doi.org/10.1063/5.0014258</li>
<li>Aho N, Buslaev P, Jansen A, et al (2022) Scalable Constant pH Molecular Dynamics in GROMACS. J Chem Theory Comput 18:6148–6160. https://doi.org/10.1021/acs.jctc.2c00516</li>
<li>Chiariello MG, Grünewald F, Zarmiento-Garcia R, Marrink SJ (2023) pH-Dependent Conformational Switch Impacts Stability of the PsbS Dimer. J Phys Chem Lett 14:905–911. https://doi.org/10.1021/acs.jpclett.2c03760</li>
<li>Claveras Cabezudo A, Athanasiou C, Tsengenes A, Wade RC (2023) Scaling Protein–Water Interactions in the Martini 3 Coarse-Grained Force Field to Simulate Transmembrane Helix Dimers in Different Lipid Environments. J Chem Theory Comput 19:2109–2119. https://doi.org/10.1021/acs.jctc.2c00950</li>
<li>Spinti JK, Neiva Nunes F, Melo MN (2023) Room for improvement in the initial martini 3 parameterization of peptide interactions. Chem Phys Lett 819:140436. https://doi.org/10.1016/j.cplett.2023.140436</li>
<li>Thomasen FE, Skaalum T, Kumar A, et al (2024) Rescaling protein-protein interactions improves Martini 3 for flexible proteins in solution. Nat Commun 15:6645. https://doi.org/10.1038/s41467-024-50647-9</li>
<li>Zerze GH (2024) Optimizing the Martini 3 Force Field Reveals the Effects of the Intricate Balance between Protein–Water Interaction Strength and Salt Concentration on Biomolecular Condensate Formation. J Chem Theory Comput 20:1646–1655. https://doi.org/10.1021/acs.jctc.2c01273</li>
<li>Koukos PI, Dehghani-Ghahnaviyeh S, Velez-Vega C, et al (2023) Martini 3 Force Field Parameters for Protein Lipidation Post-Translational Modifications. J Chem Theory Comput 19:8901–8918. https://doi.org/10.1021/acs.jctc.3c00604</li>
<li>Atsmon-Raz Y, Tieleman DP (2017) Parameterization of Palmitoylated Cysteine, Farnesylated Cysteine, Geranylgeranylated Cysteine, and Myristoylated Glycine for the Martini Force Field. J Phys Chem B 121:11132–11143. https://doi.org/10.1021/acs.jpcb.7b10175</li>
<li>Grünewald F, Punt MH, Jefferys EE, et al (2022) Martini 3 Coarse-Grained Force Field for Carbohydrates. J Chem Theory Comput 18:7555–7569. https://doi.org/10.1021/acs.jctc.2c00757</li>
<li>Brandner AF, Smith IPS, Marrink SJ, et al (2025) Systematic Approach to Parametrization of Disaccharides for the Martini 3 Coarse-Grained Force Field. J Chem Inf Model. https://doi.org/10.1021/acs.jcim.4c01874</li>
<li>Stevens JA, Grünewald F, Van Tilburg PAM, et al (2023) Molecular dynamics simulation of an entire cell. Front Chem 11:1106495. https://doi.org/10.3389/fchem.2023.1106495</li>
<li>Pezeshkian W, Grünewald F, Narykov O, et al (2023) Molecular architecture and dynamics of SARS-CoV-2 envelope by integrative modeling. Structure 31:492-503.e7. https://doi.org/10.1016/j.str.2023.02.006</li>
<li>Quemener E, Corvellec M (2013) SIDUS—the solution for extreme deduplication of an operating system. Linux J 2013:3</li>
</ol>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/cgmartini\.nl");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<div class="footer-left">

</div>
</div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://x.com/cg_martini">
      <i class="bi bi-twitter" role="img" aria-label="Martini Twitter">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="../../../../docs/faq/index.html">
<p><iconify-icon inline="true" role="img" icon="hugeicons:help-circle" style="font-size: 1.25em;" aria-label="FAQs" title="FAQs"></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/Martini-Force-Field-Initiative">
      <i class="bi bi-github" role="img" aria-label="Martini GitHub">
</i> 
    </a>
  </li>  
</ul>
    </div>
    <div class="nav-footer-right">
<div class="footer-right">
© 2025 Martini Force Field Initiative <img src="https://cgmartini-library.s3.ca-central-1.amazonaws.com/icon.png" alt="Icon">
</div>
</div>
  </div>
</footer>
<script>
    document.addEventListener('DOMContentLoaded', (event) => {
      const quotes = [
          "<i>Shaken, not stirred.</i> - James Bond",
          "<i>He was white and shaken, like a dry martini.</i> - P. G. Wodehouse",      
          "<i>I think people think we're all sipping martinis by the pool.</i> - Maria Menounos",
          "<i>Why don't you get out of that wet coat and into a dry martini?</i> - Robert Benchley",
          "<i>Really, can anyone drink several martinis at lunch?</i> - Christine Baranski",
          "<i>Zen martini: A martini with no vermouth at all. And no gin, either.</i> - P.J. O'Rourke",
          "<i>I never go jogging, it makes me spill my martini.</i> - George Burns",
          "<i>Martinis are the only American invention as perfect as a sonnet.</i> - H. L. Mencken",
          "<i>He knows just how I like my martini: full of alcohol.</i> - Homer Simpson",
          "<i>Happiness is: finding olives in your martini when you're hungry.</i> - Johnny Carson",
          "<i>If it wasn't for the olives in his martinis he'd starve to death.</i> - Milton Berle",
        ];
      const randomIndex = Math.floor(Math.random() * quotes.length);
      const randomQuote = quotes[randomIndex];

      const footerLeft = document.querySelector('.footer-left');
      if (footerLeft) {
          footerLeft.innerHTML = randomQuote;
      }
    });
</script>
<script src="../../../../js/sidebar-close.js"></script> 




</body></html>