<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.33">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Tutorial I.II - Metrics to Compare Protein Structure Models – Martini Force Field Initiative</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../../">
<link href="../../../../images/favicon.png" rel="icon" type="image/png">
<script src="../../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../../site_libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../../site_libs/bootstrap/bootstrap-094f87d8d1cb9068919611e167c728e2.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../../../site_libs/quarto-contrib/iconify-2.1.0/iconify-icon.min.js"></script>
<link href="../../../../site_libs/quarto-contrib/fontawesome6-1.2.0/all.min.css" rel="stylesheet">
<link href="../../../../site_libs/quarto-contrib/fontawesome6-1.2.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>
    document.addEventListener('DOMContentLoaded', (event) => {
      const quotes = [
          "<i>Shaken, not stirred.</i> - James Bond",
          "<i>He was white and shaken, like a dry martini.</i> - P. G. Wodehouse",      
          "<i>I think people think we're all sipping martinis by the pool.</i> - Maria Menounos",
          "<i>Why don't you get out of that wet coat and into a dry martini?</i> - Robert Benchley",
          "<i>Really, can anyone drink several martinis at lunch?</i> - Christine Baranski",
          "<i>Zen martini: A martini with no vermouth at all. And no gin, either.</i> - P.J. O'Rourke",
          "<i>I never go jogging, it makes me spill my martini.</i> - George Burns",
          "<i>Martinis are the only American invention as perfect as a sonnet.</i> - H. L. Mencken",
          "<i>He knows just how I like my martini: full of alcohol.</i> - Homer Simpson",
          "<i>Happiness is: finding olives in your martini when you're hungry.</i> - Johnny Carson",
          "<i>If it wasn't for the olives in his martinis he'd starve to death.</i> - Milton Berle",
        ];
      const randomIndex = Math.floor(Math.random() * quotes.length);
      const randomQuote = quotes[randomIndex];

      const footerLeft = document.querySelector('.footer-left');
      if (footerLeft) {
          footerLeft.innerHTML = randomQuote;
      }
    });
</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../../../styles.css">
<meta property="og:title" content="Tutorial I.II - Metrics to Compare Protein Structure Models – Martini Force Field Initiative">
<meta property="og:description" content="">
<meta property="og:image" content="https://cgmartini.nl/docs/tutorials/Martini3/ProteinsI/Fig5.png">
<meta property="og:site_name" content="Martini Force Field Initiative">
<meta property="og:image:height" content="2349">
<meta property="og:image:width" content="5128">
<meta name="twitter:title" content="Tutorial I.II - Metrics to Compare Protein Structure Models – Martini Force Field Initiative">
<meta name="twitter:description" content="">
<meta name="twitter:image" content="https://cgmartini.nl/docs/tutorials/Martini3/ProteinsI/Fig5.png">
<meta name="twitter:image-height" content="2349">
<meta name="twitter:image-width" content="5128">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../../images/martini.gif" alt="martini logo" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../../index.html"> <i class="bi bi-house-fill" role="img">
</i> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../../../docs/tutorials/index.html" aria-current="page"> <i class="bi bi-mortarboard-fill" role="img">
</i> 
<span class="menu-text">Tutorials</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/downloads/index.html"> <i class="bi bi-download" role="img">
</i> 
<span class="menu-text">Downloads</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/publications/index.html"> <i class="bi bi-newspaper" role="img">
</i> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/announcements/index.html"> 
<span class="menu-text"><i class="fa-solid fa-bullhorn" aria-label="bullhorn"></i> Announcements</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/contact/index.html"> 
<span class="menu-text"><i class="fa-regular fa-address-book" aria-label="address-book"></i> Contact</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-fa-circle-info--help" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text"><i class="fa-solid fa-circle-info" aria-label="circle-info"></i> Help</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-fa-circle-info--help">    
        <li>
    <a class="dropdown-item" href="https://github.com/orgs/Martini-Force-Field-Initiative/discussions"><i class="bi bi-chat-right-text" role="img">
</i> 
 <span class="dropdown-text">Discussion Board</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/orgs/Martini-Force-Field-Initiative/discussions"><i class="bi bi-bug" role="img">
</i> 
 <span class="dropdown-text">Report an Issue</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../../docs/faq/index.html"><i class="bi bi-question-circle" role="img">
</i> 
 <span class="dropdown-text">FAQs</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://x.com/cg_martini" title="Martini Twitter" class="quarto-navigation-tool px-1" aria-label="Martini Twitter"><i class="bi bi-twitter"></i></a>
    <a href="https://github.com/Martini-Force-Field-Initiative" title="Martini GitHub" class="quarto-navigation-tool px-1" aria-label="Martini GitHub"><i class="bi bi-github"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Tutorial I.II - Metrics to Compare Protein Structure Models</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><b>Summary</b></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/Martini3/lectures.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">🎓 Martini Lectures</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/Martini3/tutorials.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">🛠️ Hands-on Tutorials</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/Legacy/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">🕸️ Legacy Material</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#i.ii.1-root-mean-square-fluctuation-rmsf" id="toc-i.ii.1-root-mean-square-fluctuation-rmsf" class="nav-link active" data-scroll-target="#i.ii.1-root-mean-square-fluctuation-rmsf">I.II.1 Root Mean Square Fluctuation (RMSF)</a></li>
  <li><a href="#i.ii.2-root-mean-square-deviation-rmsd" id="toc-i.ii.2-root-mean-square-deviation-rmsd" class="nav-link" data-scroll-target="#i.ii.2-root-mean-square-deviation-rmsd">I.II.2 Root Mean Square Deviation (RMSD)</a></li>
  <li><a href="#i.ii.3-backbone-distance-matrix" id="toc-i.ii.3-backbone-distance-matrix" class="nav-link" data-scroll-target="#i.ii.3-backbone-distance-matrix">I.II.3 Backbone Distance Matrix</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Tutorial I.II - Metrics to Compare Protein Structure Models</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<hr>
<p>This tutorial is part of <a href="../../../../docs/tutorials/Martini3/ProteinsI/index.html"><strong><code>"Martini 3 protein models - a practical introduction to different structure bias models and their comparison"</code></strong></a><strong><code>.</code></strong></p>
<p>In case of issues, please contact <a href="mailto:%20duve@fias.uni-frankfurt.de">duve@fias.uni-frankfurt.de</a>, <a href="mailto:%20luis.borges@ens-lyon.fr">luis.borges@ens-lyon.fr</a>, or <a href="mailto:%20thallmair@fias.uni-frankfurt.de">thallmair@fias.uni-frankfurt.de</a>.</p>
If you would like to refer to information of this tutorial, please cite <a href="https://www.biorxiv.org/content/10.1101/2025.03.17.643608v2">T. Duve, L. Wang, L. Borges-Araújo, S. J. Marrink, P. C. T. Souza, S. Thallmair, Martini 3 Protein Models - A Practical Introduction to Different Structure Bias Models and their Comparison, bioRxiv (2025), doi: 10.1101/2025.03.17.643608</a>.
<hr>
<p>This tutorial Section discusses different different metrics that can be used to judge which protein structure model is the best suited for a specific use case.</p>
<section id="i.ii.1-root-mean-square-fluctuation-rmsf" class="level2">
<h2 class="anchored" data-anchor-id="i.ii.1-root-mean-square-fluctuation-rmsf">I.II.1 Root Mean Square Fluctuation (RMSF)</h2>
<p>The Root Mean Square Fluctuation (RMSF) is a metric that quantifies the fluctuation of the position of an atom over time. It is calculated according to the following equation, where <span class="math inline">\(r_i\)</span> is the coordinate vector of a particle <span class="math inline">\(i\)</span>, <span class="math inline">\(\bar{r}_i\)</span> the ensemble average coordinate vector of <span class="math inline">\(i\)</span> and <span class="math inline">\(T\)</span> the total number of frames in the trajectory.</p>
<p><span class="math display">\[
RMSF(i) = \sqrt{\frac{1}{T} \sum_{t=1}^{T} \left( r_i(t) - \bar{r}_i \right)^2}
\]</span></p>
<p>The RMSF can be used to quantify the flexibility of a structure over the course of a simulation. Regions with high RMSF values indicate high structural mobility, while regions with low RMSF values are more rigid.</p>
<p>To compare the RMSF between two simulations, e.g.&nbsp;an atomistic reference and the CG counterpart, the ΔRMSF can be calculated with the following equation. An ideal model would have a ΔRMSF of zero for all residues.</p>
<p><span class="math display">\[
\Delta RMSF = RMSF_{CG} - RMSF_{atomistic}
\]</span></p>
<p>In our case we use a <a href="https://github.com/cristina-gil-herrero/RMSFscript">script to calculate the RMSF</a> of the C<span class="math inline">\(_α\)</span>-atoms or backbone beads for atomistic and CG trajectories, respectively. First, the atomistic trajectories are divided into 200 ns segments, while CG trajectories are divided into 50 ns segments, following the 1:4 ratio recommended for comparisons between atomistic and CG Martini simulations, due to the smoothed energy landscape of CG simulations[34]. Subsequently, the protein is centred in the simulation box. In seven iterations, the trajectory is fitted to all C<span class="math inline">\(_α\)</span>-atoms (atomistic) or BB beads (CG) with a mean RMSF value below the threshold of 1.5 Å in the previous iteration, using gmx trjconv with the flag -fit rot+trans. This is done to avoid fitting the trajectory to highly flexible regions of the protein. The final mean RMSF values for each residue are calculated following the seventh iteration. RMSF values are computed using gmx rmsf with the flags -nofit -res.</p>
<p>The RMSF of a protein is a key metric to judge the viability of a protein structure model, as it can indicate to what extent a certain model can reproduce the intrinsic flexibility of a protein. In the following Sections, we will first use it to determine the optimal parameters for the EN and GōMartini models. Then, we will compare these optimized models to the OLIVES model.</p>
<section id="i.ii.1.1-tuning-of-the-en-and-gōmartini-models" class="level4">
<h4 class="anchored" data-anchor-id="i.ii.1.1-tuning-of-the-en-and-gōmartini-models">I.II.1.1 Tuning of the EN and GōMartini Models</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Fig5.png" class="img-fluid figure-img"></p>
<figcaption>Figure 5: Backbone RMSF deviation (ΔRMSF) between the CG models and the atomistic reference of CK1δ using different parameters for the EN and GōMartini models. A: EN model with k<span class="math inline">\(_B\)</span> = 700, 800, and 1000 kJ/(mol nm<span class="math inline">\(^2\)</span> <em>). B: GōMartini model with ε</em><span class="math inline">\(_{LJ}\)</span> = 8, 10, 12, and 14 kJ/mol.</figcaption>
</figure>
</div>
<p>Figure 5 depicts the ΔRMSF for the different parameters of the EN and GōMartini models. We tested three values for the force constant k<span class="math inline">\(_B\)</span> of the EN model: 700, 800, and 1000 kJ/(mol nm<span class="math inline">\(^2\)</span> ). Overall, only minor changes in the RMSF of the protein can be observed. As expected, a higher force constant decreases the overall flexibility of the protein and therefore the RMSF. Figure 5A shows that the higher force constants, particularly 1000 kJ/(mol nm<span class="math inline">\(^2\)</span> ), lead to an underestimation of the flexibility of CK1δ. Therefore, we will proceed with the force constant value of 700 kJ/(mol nm<span class="math inline">\(^2\)</span> ) in the following steps. For the GōMartini model, four values for the potential depth <em>ε</em><span class="math inline">\(_{LJ}\)</span> were tested: 8, 10, 12, and 14 kJ/mol. Figure 5B shows that the potential depth of the GōMartini model has a substantial impact on the RMSF of CK1δ. Overall, higher values for <em>ε</em><span class="math inline">\(_{LJ}\)</span> better capture the flexibility of CK1δ, both in more rigid regions, as well as in highly flexible loops. Therefore, we will proceed with a potential depth of 14 kJ/mol.</p>
</section>
<section id="i.ii.1.2-comparison-of-the-structure-bias-models" class="level4">
<h4 class="anchored" data-anchor-id="i.ii.1.2-comparison-of-the-structure-bias-models">I.II.1.2 Comparison of the Structure Bias Models</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Fig6.png" class="img-fluid figure-img"></p>
<figcaption>Figure 6: Comparison of the backbone RMSF of CK1δ between protein structure models (EN, GōMartini, OLIVES and OLIVES + DSSP). Left: RMSF distribution over all residues and time windows. Right: RMSF of individual residues. The atomistic reference is displayed in black. Solid lines show the mean of all time windows, the shaded area the mean ± standard deviation. Residues missing in the reference crystal structure are indicated by the shaded rectangles.</figcaption>
</figure>
</div>
<p>When comparing the C<span class="math inline">\(_α\)</span>/BB RMSF of different structure bias models, examine the RMSF on a per-residue basis to identify deviations in specific regions, paying attention to whether the flexibility of the more rigid regions is reproduced. Flexible regions, such as loops and termini, while important, are more challenging to reproduce. Typically, more deviation from the reference is accepted here for the overall evaluation. Figure 6 shows the RMSF distribution and RMSF for each residue in comparison to the atomistic reference simulation (as opposed to the ΔRMSF in Figure 5). This allows us to identify highly flexible and more rigid regions of CK1δ. Depending on the scientific question at hand, it might be useful to focus the analysis on certain regions of the protein, which are important for the specific system, e.g.&nbsp;a ligand binding pocket or a protein-protein interface.</p>
<p>With a mean RMSF of 0.91 Å, CK1δ is a rather rigid protein, favouring more rigid protein models, such as the EN model and GōMartini model with a high potential depth. Figure 6 shows that both the EN and GōMartini model overall capture the flexibility of CK1δ very well, especially in the more rigid regions. However, both models slightly overestimate the RMSF of residues 260-265, while underestimating the flexibility of the loop in the region 171-175. The GōMartini model shows an increased RMSF in the residues 163-171 located between two loops.</p>
<p>In contrast, the OLIVES model consistently overestimates the flexibility of the more rigid regions of CK1δ, but also two of the loops that are missing in the crystal structure (residues 43-46 and 217-222). The addition of DSSP secondary structure restraints resulting in the OLIVES + DSSP model reduces this overestimation and reproduces the flexibility of the rigid region between residues 50-150 and the two loops (residues 43-46 and 217-222) well. However, while improved compared to the OLIVES model, the RMSF of the rigid region between the residues 225-285 is still overestimated with the OLIVES + DSSP model. Like the GōMartini model, both models overestimate the RMSF of residues 163-171. For the OLIVES model, a scaling of the interaction strength is also possible (see <a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut1.html">Section I.I.3.4</a>). This might improve the comparison to the atomistic reference.</p>
<p>To conclude, in this case both the EN and GōMartini models demonstrate strong performance in reproducing the RMSF of the atomistic reference simulation, with the EN model showing a slight edge. These models especially capture the dynamics of rigid regions most accurately. The combination of the OLIVES model and DSSP restraints provides an acceptable performance, although less precise than the EN and GōMartini models, due to an overall overestimation of the flexibility of CK1δ. The OLIVES model alone struggles to accurately reflect the RMSF of CK1δ, especially in rigid areas.</p>
</section>
</section>
<section id="i.ii.2-root-mean-square-deviation-rmsd" class="level2">
<h2 class="anchored" data-anchor-id="i.ii.2-root-mean-square-deviation-rmsd">I.II.2 Root Mean Square Deviation (RMSD)</h2>
<p>The Root Mean Square Deviation (RMSD) is a measure used to quantify structural similarity, typically between a frame of a simulation and a reference structure, such as a crystal structure. To calculate the RMSD along a trajectory, at each frame, the structure is aligned to the reference structure. Then, the RMSD is calculated according to the following equation, <span class="math inline">\(r_i(t)\)</span> are the coordinates of particle <span class="math inline">\(i\)</span> at frame <span class="math inline">\(t\)</span>, and <span class="math inline">\(r_i^{\mathrm{ref}}\)</span> the coordinates of particle i in the reference structure, and N the total number of particles.</p>
<p><span class="math display">\[
RMSD(t) = \sqrt{\frac{1}{N} \sum_{i=1}^{N} \left( r_i(t) - r_i^{ref} \right)^2}
\]</span></p>
<p>One might notice the similarity between this equation and the equation to calculate the RMSF (see Section I.II.1). While both metrics have a similar concept, they contain different information. The RMSD provides a global comparison, assessing overall structural deviations relative to a reference structure over time. In contrast, the RMSF is calculated relative to an averaged structure and measures the flexibility of individual residues over the course of a trajectory. While the RMSD evaluates global structure deviation, the RMSF provides insight into local dynamics and flexibility.</p>
<p>In our case, we calculate the RMSD of all C<span class="math inline">\(_α\)</span>-atoms (atomistic) or BB beads (CG) along our trajectories. First, we centre the protein in the simulation box. Additionally, we create an index file that contains the indices of all C<span class="math inline">\(_α\)</span>-atoms (atomistic) or BB beads (CG). Then we use this index file and gmx rms to calculate the RMSD. The command will prompt you to first select the index group on which to fit the trajectory, then which index group to use for the calculation of the RMSD. We select CA/BB for both.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">gmx</span> trjconv <span class="at">-f</span> production.xtc <span class="at">-pbc</span> whole <span class="at">-center</span> <span class="at">-s</span> production.tpr <span class="at">-o</span> noPBC.xtc </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">gmx</span> make_ndx <span class="at">-f</span> equilibration.gro <span class="at">-o</span> index.ndx gmx rms <span class="at">-f</span> noPBC.xtc <span class="at">-s</span> production.tpr <span class="at">-n</span> CA.ndx <span class="at">-o</span> rmsd.xvg </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>It might be useful to also consider other selections for the RMSD calculation. For example, one could calculate the RMSD of a ligand bound to a binding pocket relative to a crystal structure, to judge the ability of a model to reproduce experimentally observed binding modes. The RMSD of the protein can also be used to judge if the system is well equilibrated. For an equilibrated system, one expects a converged RMSD time series.</p>
<section id="i.ii.2.1-the-unbiased-coarse-grained-model" class="level4">
<h4 class="anchored" data-anchor-id="i.ii.2.1-the-unbiased-coarse-grained-model">I.II.2.1 The Unbiased Coarse-Grained Model</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Fig7.png" class="img-fluid figure-img"></p>
<figcaption>Figure 7: Unbiased CG Model. A: Example RMSD traces over time of atomistic reference and CG protein structure models. B,C: Snapshot of unbiased CG model at the beginning of the simulation (B) and after 0.9 μs effective time (C).</figcaption>
</figure>
</div>
<p>Figure 7 illustrates the need for the structure bias layer in Martini 3. While the atomistic trajectory and biased CG trajectories show RMSD values of below 4.5 Å, the unbiased model exhibits values of up to 23 Å. This corresponds to the unfolding of the protein depicted in Figure 7C, showing that the Martini 3 protein model cannot retain a tertiary structure without additional biasing.</p>
</section>
<section id="i.ii.2.2-comparison-of-the-structure-bias-models" class="level4">
<h4 class="anchored" data-anchor-id="i.ii.2.2-comparison-of-the-structure-bias-models">I.II.2.2 Comparison of the Structure Bias Models</h4>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Fig8.png" class="img-fluid figure-img" style="width:50.0%"></p>
<figcaption>Figure 8: RMSD distributions of CK1δ. Atomistic used C<span class="math inline">\(_α\)</span>-atoms and CG models used BB beads to fit the trajectory and calculate the RMSD.</figcaption>
</figure>
</div>
<p>Figure 8 shows the RMSD distributions of the atomistic reference and the respective structure bias models for CK1δ. We see that the atomistic reference distribution ranges from 1.5 – 3.0 Å. The OLIVES model shows the distribution with the highest deviation from the atomistic distribution, ranging from 2.5 – 4.5 Å. In agreement with previous results, the addition of DSSP restraints to the OLIVES model leads to a decrease in the RMSD and higher agreement with the atomistic reference. The GōMartini shows even higher overlap with the atomistic reference, resulting in the best estimation of all models here, while also showing higher RMSD values up to 3.5 Å not present in the atomistic reference. The EN model shows the lowest RMSD values of all models, with values ranging from 1.5 – 2.5 Å. While the distribution overlaps with the atomistic reference distribution, the model is unable to reproduce the higher RMSD values also present in the atomistic reference. This indicates the strong biasing of the EN model towards the crystal structure that was used to create the model.</p>
</section>
</section>
<section id="i.ii.3-backbone-distance-matrix" class="level2">
<h2 class="anchored" data-anchor-id="i.ii.3-backbone-distance-matrix">I.II.3 Backbone Distance Matrix</h2>
<p>The final metric we are going to analyse is the distances between the backbone beads of CK1δ. Using a nested loop and <code>gmx distance</code> we can calculate this distance for each residue pair.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">gmx</span> distance <span class="at">-f</span> noPBC.xtc <span class="at">-s</span> production.tpr <span class="at">-select</span> <span class="st">"resid </span><span class="va">${i}</span><span class="st"> </span><span class="va">${j}</span><span class="st"> and name BB"</span> <span class="at">-len</span> 4.0 <span class="at">-binw</span> 0.01 <span class="at">-oh</span> <span class="va">${i}</span>-<span class="va">${j}</span>_BB-dist.xvg </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>For the reference atomistic trajectory we must ensure that the distance is calculated between where the BB bead would be located, at the center of mass (COM) of the backbone atoms. Using the flag <code>-select "resid ${i} ${j} and name N CA C O"</code> we select all non-hydrogen atoms of the protein backbone. We then specify that the COM should be used for the calculation using <code>-selrpos part_res_com</code> and <code>-seltype part_res_com</code>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">gmx</span> distance <span class="at">-f</span> noPBC.xtc <span class="at">-s</span> production.tpr <span class="at">-select</span> <span class="st">"resid </span><span class="va">${i}</span><span class="st"> </span><span class="va">${j}</span><span class="st"> and name N CA C O"</span> <span class="at">-selrpos</span> part_res_com <span class="at">-seltype</span> part_res_com <span class="at">-len</span> 4.0 <span class="at">-binw</span> 0.01 <span class="at">-oh</span> <span class="va">${i}</span>-<span class="va">${j}</span>_BB-dist.xvg </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>This returns a histogram of distances for all residue pairs, ranging from 0 to 80 Å (specified by <code>-len 4.0</code>). Note that this approach generates a large number of files, so check your storage availability and delete the created files after performing the necessary calculations.</p>
<p>We can now calculate the mean of each histogram and plot them in a matrix. To judge the similarity of the distance distributions of a protein structure model to the atomistic reference, we can calculate the integrated absolute difference [35]. For this, we calculate the absolute difference of two histograms and sum up the values. Identical distributions then have a sum of zero, while distributions with no overlap sum up to two. Figure 9 shows both metrics for all models of CK1δ. Similarly to the analysis of the RMSF, here, the focus should be rather on rigid regions or regions of interest, such as potential ligand binding sites.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Fig9.png" class="img-fluid figure-img"></p>
<figcaption>Figure 9: Matrix representation of backbone bead to backbone bead distances for all residues. Displayed is the mean distance and integrated absolute difference.</figcaption>
</figure>
</div>
<p>As we saw with the RMSD in Figure 7, the CG model without a structural bias shows larger distances than the atomistic reference. This is the case for all residue pairs, expect for individual pairs near the diagonal, which represents very short-range interactions mediated by the DSSP restraints. This shows that while the protein structure largely unfolds during the simulation, some secondary structure motifs remain intact.</p>
<p>Notably, in all models the distance pairs including the missing loop in the crystal structure (residues 171-173) show large deviations from the atomistic reference for all models. This is in line with the deviations of the RMSF in this region that was observed for all models (see Section I.II.1.2). Interestingly, all models generally displayed deviations between residues 175 and 200, although the EN and GōMartini reproduce the flexibility of this region well. A comparison between the OLIVES and OLIVES + DSSP models reveals that OLIVES + DSSP produces better agreement at distance pairs along the diagonal, corresponding to short-range interactions. This indicates a better representation of the secondary structure of CK1δ. In terms of overall distance accuracy, the GōMartini model outperforms the others, displaying the best overall fit. The highest deviations in distances are within loop regions.</p>
<hr>
<p>The following section (<a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut3.html">I.III</a>) will discuss how to best set up IDRs in Martini 3 proteins.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/cgmartini\.nl");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<div class="footer-left">

</div>
</div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://x.com/cg_martini">
      <i class="bi bi-twitter" role="img" aria-label="Martini Twitter">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="../../../../docs/faq/index.html">
<p><iconify-icon inline="true" role="img" icon="hugeicons:help-circle" style="font-size: 1.25em;" aria-label="FAQs" title="FAQs"></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/Martini-Force-Field-Initiative">
      <i class="bi bi-github" role="img" aria-label="Martini GitHub">
</i> 
    </a>
  </li>  
</ul>
    </div>
    <div class="nav-footer-right">
<div class="footer-right">
© 2025 Martini Force Field Initiative <img src="https://cgmartini-library.s3.ca-central-1.amazonaws.com/icon.png" alt="Icon">
</div>
</div>
  </div>
</footer>
<script>
    document.addEventListener('DOMContentLoaded', (event) => {
      const quotes = [
          "<i>Shaken, not stirred.</i> - James Bond",
          "<i>He was white and shaken, like a dry martini.</i> - P. G. Wodehouse",      
          "<i>I think people think we're all sipping martinis by the pool.</i> - Maria Menounos",
          "<i>Why don't you get out of that wet coat and into a dry martini?</i> - Robert Benchley",
          "<i>Really, can anyone drink several martinis at lunch?</i> - Christine Baranski",
          "<i>Zen martini: A martini with no vermouth at all. And no gin, either.</i> - P.J. O'Rourke",
          "<i>I never go jogging, it makes me spill my martini.</i> - George Burns",
          "<i>Martinis are the only American invention as perfect as a sonnet.</i> - H. L. Mencken",
          "<i>He knows just how I like my martini: full of alcohol.</i> - Homer Simpson",
          "<i>Happiness is: finding olives in your martini when you're hungry.</i> - Johnny Carson",
          "<i>If it wasn't for the olives in his martinis he'd starve to death.</i> - Milton Berle",
        ];
      const randomIndex = Math.floor(Math.random() * quotes.length);
      const randomQuote = quotes[randomIndex];

      const footerLeft = document.querySelector('.footer-left');
      if (footerLeft) {
          footerLeft.innerHTML = randomQuote;
      }
    });
</script>
<script src="../../../../js/sidebar-close.js"></script> 




</body></html>