<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Tutorial I.III - Simulations of IDRs in Martini 3 ‚Äì Martini Force Field Initiative</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../../">
<link href="../../../../images/favicon.png" rel="icon" type="image/png">
<script src="../../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../../site_libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../../site_libs/bootstrap/bootstrap-69fe911abf326d3c3f52f7dd89351358.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../../../../site_libs/quarto-contrib/iconify-2.1.0/iconify-icon.min.js"></script>
<link href="../../../../site_libs/quarto-contrib/fontawesome6-1.2.0/all.min.css" rel="stylesheet">
<link href="../../../../site_libs/quarto-contrib/fontawesome6-1.2.0/latex-fontsize.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script>
    document.addEventListener('DOMContentLoaded', (event) => {
      const quotes = [
          "<i>Shaken, not stirred.</i> - James Bond",
          "<i>He was white and shaken, like a dry martini.</i> - P. G. Wodehouse",      
          "<i>I think people think we're all sipping martinis by the pool.</i> - Maria Menounos",
          "<i>Why don't you get out of that wet coat and into a dry martini?</i> - Robert Benchley",
          "<i>Really, can anyone drink several martinis at lunch?</i> - Christine Baranski",
          "<i>Zen martini: A martini with no vermouth at all. And no gin, either.</i> - P.J. O'Rourke",
          "<i>I never go jogging, it makes me spill my martini.</i> - George Burns",
          "<i>Martinis are the only American invention as perfect as a sonnet.</i> - H. L. Mencken",
          "<i>He knows just how I like my martini: full of alcohol.</i> - Homer Simpson",
          "<i>Happiness is: finding olives in your martini when you're hungry.</i> - Johnny Carson",
          "<i>If it wasn't for the olives in his martinis he'd starve to death.</i> - Milton Berle",
        ];
      const randomIndex = Math.floor(Math.random() * quotes.length);
      const randomQuote = quotes[randomIndex];

      const footerLeft = document.querySelector('.footer-left');
      if (footerLeft) {
          footerLeft.innerHTML = randomQuote;
      }
    });
</script>


<link rel="stylesheet" href="../../../../styles.css">
<meta property="og:title" content="Tutorial I.III - Simulations of IDRs in Martini 3 ‚Äì Martini Force Field Initiative">
<meta property="og:description" content="">
<meta property="og:image" content="https://cgmartini.nl/docs/tutorials/Martini3/ProteinsI/Fig10.png">
<meta property="og:site_name" content="Martini Force Field Initiative">
<meta property="og:image:height" content="1378">
<meta property="og:image:width" content="3346">
<meta name="twitter:title" content="Tutorial I.III - Simulations of IDRs in Martini 3 ‚Äì Martini Force Field Initiative">
<meta name="twitter:description" content="">
<meta name="twitter:image" content="https://cgmartini.nl/docs/tutorials/Martini3/ProteinsI/Fig10.png">
<meta name="twitter:image-height" content="1378">
<meta name="twitter:image-width" content="3346">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-sidebar floating nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../../images/martini.gif" alt="martini logo" class="navbar-logo">
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../../../index.html"> <i class="bi bi-house-fill" role="img">
</i> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="../../../../docs/tutorials/index.html" aria-current="page"> <i class="bi bi-mortarboard-fill" role="img">
</i> 
<span class="menu-text">Tutorials</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/downloads/index.html"> <i class="bi bi-download" role="img">
</i> 
<span class="menu-text">Downloads</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/publications/index.html"> <i class="bi bi-newspaper" role="img">
</i> 
<span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/announcements/index.html"> 
<span class="menu-text"><i class="fa-solid fa-bullhorn" aria-label="bullhorn"></i> Announcements</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../../docs/contact/index.html"> 
<span class="menu-text"><i class="fa-regular fa-address-book" aria-label="address-book"></i> Contact</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-fa-circle-info--help" role="link" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text"><i class="fa-solid fa-circle-info" aria-label="circle-info"></i> Help</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-fa-circle-info--help">    
        <li>
    <a class="dropdown-item" href="https://github.com/orgs/Martini-Force-Field-Initiative/discussions"><i class="bi bi-chat-right-text" role="img">
</i> 
 <span class="dropdown-text">Discussion Board</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="https://github.com/orgs/Martini-Force-Field-Initiative/discussions"><i class="bi bi-bug" role="img">
</i> 
 <span class="dropdown-text">Report an Issue</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../../../../docs/faq/index.html"><i class="bi bi-question-circle" role="img">
</i> 
 <span class="dropdown-text">FAQs</span></a>
  </li>  
    </ul>
  </li>
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
    <a href="https://x.com/cg_martini" title="Martini Twitter" class="quarto-navigation-tool px-1" aria-label="Martini Twitter"><i class="bi bi-twitter"></i></a>
    <a href="https://github.com/Martini-Force-Field-Initiative" title="Martini GitHub" class="quarto-navigation-tool px-1" aria-label="Martini GitHub"><i class="bi bi-github"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Tutorial I.III - Simulations of IDRs in Martini 3</li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><b>Summary</b></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/Martini3/lectures.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">üéì Martini Lectures</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/Martini3/tutorials.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">üõ†Ô∏è Hands-on Tutorials</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../docs/tutorials/Legacy/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">üï∏Ô∏è Legacy Material</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#i.iii.1-adding-a-water-bias" id="toc-i.iii.1-adding-a-water-bias" class="nav-link active" data-scroll-target="#i.iii.1-adding-a-water-bias">I.III.1 Adding a Water Bias</a></li>
  <li><a href="#i.iii.2-martini3-idp" id="toc-i.iii.2-martini3-idp" class="nav-link" data-scroll-target="#i.iii.2-martini3-idp">I.III.2 Martini3-IDP</a></li>
  <li><a href="#i.iii.2-simulation-of-the-multi-domain-ck1Œ¥-model" id="toc-i.iii.2-simulation-of-the-multi-domain-ck1Œ¥-model" class="nav-link" data-scroll-target="#i.iii.2-simulation-of-the-multi-domain-ck1Œ¥-model">I.III.2 Simulation of the Multi-Domain CK1Œ¥ Model</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Tutorial I.III - Simulations of IDRs in Martini 3</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<hr>
<p>This tutorial is part of <a href="../../../../docs/tutorials/Martini3/ProteinsI/index.html"><strong><code>"Martini 3 protein models - a practical introduction to different structure bias models and their comparison"</code></strong></a><strong><code>.</code></strong></p>
<p>In case of issues, please contact <a href="mailto:%20duve@fias.uni-frankfurt.de">duve@fias.uni-frankfurt.de</a>, <a href="mailto:%20luis.borges@ens-lyon.fr">luis.borges@ens-lyon.fr</a>, or <a href="mailto:%20thallmair@fias.uni-frankfurt.de">thallmair@fias.uni-frankfurt.de</a>.</p>
If you would like to refer to information of this tutorial, please cite <a href="https://www.biorxiv.org/content/10.1101/2025.03.17.643608v2">T. Duve, L. Wang, L. Borges-Ara√∫jo, S. J. Marrink, P. C. T. Souza, S. Thallmair, Martini 3 Protein Models - A Practical Introduction to Different Structure Bias Models and their Comparison, bioRxiv (2025), doi: 10.1101/2025.03.17.643608</a>.
<hr>
<p>This section focusses on introducing the IDR of CK1Œ¥ to the system. Previous works have shown, that simulations of IDRs using the default Martini 3.0.0 force field result in too compact IDR conformations and an underestimation of the radius of gyration [10, 13, 36, 37]. Therefore, the force field should be modified to account for the specific properties of IDRs. Here, we present two options: modifying the BB-water interaction strength and the dedicated force field Martini3-IDP.</p>
<section id="i.iii.1-adding-a-water-bias" class="level2">
<h2 class="anchored" data-anchor-id="i.iii.1-adding-a-water-bias">I.III.1 Adding a Water Bias</h2>
<p>One way to improve the performance of Martini 3 when simulating IDRs is to modify the strength of the protein-water interactions for the IDR only. This technique utilizes the BB virtual sites, which are also used in the G≈çMartini model to implement the structural bias, to modify the LJ potential depth between the BB and water beads. Since these virtual sites overlap with the BB bead, the non-bonded interactions between BB and water beads can be effectively adjusted by introducing an additional potential between the virtual site and water beads [13].</p>
<p>In this approach to set up the multi-domain protein model, we use the full length CK1Œ¥ atomistic structure as Martinize2 input. As the IDR structure is typically not present in experimental structures, AlphaFold or SWISS-MODEL can be used to generate the full length atomistic protein structure (<code>CK1d_IDR.pdb</code>)[23, 24]. In Martinize2, the water bias can be added via the <code>-water-bias</code> flag. The residues of the IDR are defined via <code>-id-regions 293:415</code>, and the strength of the water-bias is set to a value of 0.5 kJ/mol here by <code>-water-bias-eps idr:0.5</code> [13]. Additionally, we need to manually specify the secondary structure of our protein with the flag <code>-ss</code> using the string returned by DSSP in <a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut1.html">Section I.I.3.1</a>. To avoid creating unwanted secondary structures in the IDR of the protein, which are sometimes wrongfully present in the atomistic reference structure, replace all characters of the IDR with <code>‚ÄòC‚Äô</code> for coil. The water bias can be combined with the G≈çMartini model for the structured region, although any of the protein structure models discussed previously are also compatible. Here we add an EN for the structured region with <code>-eunit 1:292</code>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">martinize2</span> <span class="at">-f</span> CK1d_IDR.pdb <span class="at">-x</span> CK1d_cg.pdb <span class="at">-o</span> CK1d_only.top <span class="at">-ff</span> martini3001 </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">-p</span> backbone <span class="at">-ss</span> CCCEETTTEEEEEEEEEETTEEEEEEEETTTTEEEEEEEEETTSSSCCHHHHHHHHHHHTTCTTCCE</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="ex">EEEEEETTEEEEEEECCCCBHHHHHHHTTTCCCHHHHHHHHHHHHHHHHHHHHTTEECSCCCGGGEEECCGGGTTCEEECCGGGCEECBCTTTCCBCCCCCSCCCCSCTTTCCHHHHTTCCCCHHHHHHHHHHHHHHHHHSSCTTSSCCCSSHHHHHHHHHHHHHTSCHHHHTTTSCHHHHHHHHHHHHCCTTCCCCHHHHHHHHHHHHHHTTCCCSCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC</span> <span class="at">-elastic</span> <span class="at">-el</span> 0 <span class="at">-eu</span> 0.85 <span class="at">-eunit</span> 1:292 <span class="at">-water-bias</span> <span class="at">-water-bias-eps</span> idr:0.5 <span class="at">-id-regions</span> 293:415  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>In addition to the CG protein structure and topology, Martinize2 returns two files: virtual_sites_atomtypes.itp, and virtual_sites_nonbond_params.itp, which need to be added to the Martini 3 topology file, as described in <a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut1.html">Section I.I.3.3</a>.</p>
</section>
<section id="i.iii.2-martini3-idp" class="level2">
<h2 class="anchored" data-anchor-id="i.iii.2-martini3-idp">I.III.2 Martini3-IDP</h2>
<p>Another way to improve the performance of IDRs in Martini 3 is offered by the recently developed Martini 3 disordered protein force field (Martini3-IDP)[38], which is well integrated into the current Martini 3 framework. All force field files and Python scripts in this tutorial Section are available on <a href="https://github.com/Martini-Force-Field-Initiative/Martini3-IDP-parameters">GitHub</a>.</p>
<p>Currently, two methods to generate the Martini3-IDP model are available.</p>
<section id="i.iii.2.1-setup-using-martinize2-and-polyply" class="level4">
<h4 class="anchored" data-anchor-id="i.iii.2.1-setup-using-martinize2-and-polyply">I.III.2.1 Setup Using Martinize2 and Polyply</h4>
<p>In this approach, the Martini model of a folded domain and the IDR are first generated separately with Martinize2 and Polyply. In a second step, both models are merged.</p>
<p>As in Section 2.3.2, the Martini model of the CK1Œ¥ protein folded domain with EN bias is generated by Martinize2 as following. The EN bias is used in this tutorial Section as example, but G≈çMartini and OLIVES could also be transferred easily referring to the tutorial in <a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut1.html">Section I.I.3</a>.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">martinize2</span> <span class="at">-f</span> CK1d.pdb <span class="at">-x</span> CK1d_cg.pdb <span class="at">-o</span> CK1d_only.top <span class="at">-ff</span> martini3001 <span class="at">-p</span> backbone <span class="at">-dssp</span> <span class="at">-elastic</span> <span class="at">-el</span> 0 <span class="at">-eu</span> 0.85</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Polyply is used to generate the topology and coordinates files of the IDR and can be obtained from <a href="https://github.com/marrink-lab/polyply_1.0">GitHub</a>[39]. <code>polyply gen_params</code> is used to generate the topology with protein sequence fasta input by <code>-seqf</code> (or protein sequence string by <code>-seq</code>). The <code>-f Martini3-IDP_Polyply.ff</code> flag provides the force field. <code>polyply gen_coords</code> is used to generate random IDR coordinates file by a random walk. As input, only the IDR topology is required (<code>-p</code> flag). The flag <code>-box</code> defines the edge length of the simulation box, <code>-o</code> defines the output file name containing the coordinates.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">polyply</span> gen_params <span class="at">-name</span> IDR <span class="at">-f</span> Martini3-IDP_Polyply.ff <span class="at">-seqf</span> IDR.fasta <span class="at">-o</span> IDR.itp</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="ex">polyply</span> gen_coords <span class="at">-p</span> topol.top <span class="at">-name</span> IDR <span class="at">-box</span> 30 30 30 <span class="at">-o</span> IDR.gro</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Note that the topology provided with the <code>-p</code> flag is not the <code>.itp</code> file of the IDR, but the GROMACS <code>.top</code> file including all necessary force field files involved in IDR.itp, similar to this:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#include "./martini_v3.0.0.itp"</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co">#include "./martini_v3.0.0_solvents_v1.itp"</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co">#include "./martini_v3.0.0_ions_v1.itp"</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co">#include "IDR.itp"</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="bu">[</span> molecules <span class="bu">]</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="ex">IDR</span>    1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Finally, a Python script <code>topology_merging_general.py</code> is used to merge all domains‚Äô <code>.itp</code> files and define the parameters of the crosslinking region. You must change the <code>.itp</code> file name and molecule type to <code>‚Äúmol0 mol1 ‚Ä¶‚Äù</code> according to their order in the protein. And because sometimes dihedral parameters in Martini topology are defined in two individual <code>[ dihedrals ]</code> derivatives, for consistency, we need to merge <code>[ dihedrals ]</code> derivatives in the <code>.itp</code> files with two <code>[ dihedrals ]</code> derivatives by definition in <code>topology_merging_general.py</code>.</p>
<p>Then based on the domain number in the protein, we need to adjust some definitions by order in <code>topology_merging_general.py</code> (For example, two domains here in CK1Œ¥ protein <code>‚Äúmol0-mol1‚Äù</code> are merged).</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mol0_BB,</span> mol0_SC, mol1_BB, mol1_SC = entry_edit<span class="er">(</span><span class="ex">mols,</span> <span class="st">'mol0'</span>, <span class="st">'mol1'</span><span class="kw">)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="ex">interaction_adding</span><span class="er">(</span><span class="ex">ff,</span> molname, mol0_BB, mol0_SC, mol1_BB, mol1_SC<span class="kw">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Because the protein CK1Œ¥ contains only one folded domain and one IDR, <code>‚Äòmol0-mol1‚Äô</code> is specified with the <code>-i</code> flag. The default name of the output <code>.itp</code> file is <code>mol.itp</code>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">python3</span> topology_merging_general.py <span class="at">-i</span> mol0-mol1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Note that the terminal bead type in each domain fragment (i.e.&nbsp;the crosslinked residues in the final multi-domain protein) must be checked and adjusted manually to the default Martini 3 backbone bead for that particular residue in the output <code>mol.itp</code> (see Figure 1).</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">714</span> P2   292 MET BB  714  0.0</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a> <span class="ex">‚Åù</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="ex">716</span> P2   293 LEU BB  716  0.0  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>To obtain the coordinate file of the complete multi-domain protein, insert separate domains structures into one box, then after a short simulation, the broken domain fragments should be connected due to the bonded linking between fragments in mol.itp.</p>
</section>
<section id="i.iii.2.1-setup-using-martinize2-only" class="level4">
<h4 class="anchored" data-anchor-id="i.iii.2.1-setup-using-martinize2-only">I.III.2.1 Setup Using Martinize2 only</h4>
<p>A Martinize2 format Martini3-IDP force field file is also provided, which deals with the differences between the standard Martini 3 protein model and Martini3-IDP model. For example, <code>-scfix</code> is applied in the Martini 3 protein model to define the sidechain orientation relying on atomistic reference structure, whereas general bonded parameters are used to define sidechain orientation in Martini3-IDP independent on the atomistic reference structure.</p>
<p>A <a href="https://github.com/Martini-Force-Field-Initiative/Martini3-IDP-parameters/tree/main/force%20filed/Martinize2">Martinize2 format Martini3-IDP force field file</a> is provided, which deals with the construction discrepancy between standard Martini3 protein and Martini3-IDP. To integrate Martini3-IDP into Martinize2, copy the provided <code>/force_fields</code> and <code>/mappings</code> directories to <code>/vermouth/data/force_fields/</code> and <code>/vermouth/data/mappings/</code>, respectively, and rename them as <code>Martini3-IDP</code>.</p>
<p>Similar to Section 4.1, the EN bias is generated only for the folded domain using the <code>-eunit flag</code>. To define the IDR easily, a new secondary structure <code>‚ÄòD‚Äô</code> is provided in the <code>-ss</code> flag. Note that <code>‚ÄòD‚Äô</code> is not a pre-set secondary structure, so <code>‚ÄòD‚Äô</code> secondary structure should be manually added to <code>‚Äòvermouth/dssp/dssp.py‚Äô</code> <code>ss_cg</code> list.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">martinize2</span> <span class="at">-f</span> CK1d_IDR.pdb <span class="at">-x</span> CK1d_cg.pdb <span class="at">-o</span> CK1d_only.top <span class="at">-ff</span> martini3001-IDP </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="ex">-p</span> backbone <span class="at">-ss</span> CCCEETTTEEEEEEEEECSSSEEEEEEETTTTEEEEEEEEETTCSSCCHHHHHHHHHHHTTSTTCCCEEEEEEETTEEEEEEECCCCBHHHHHHHTTTCCCHHHHHHHHHHHHHHHHHHHHTTEECCCCCGGGEEECCGGGTTCEEECCCTTCEECBCTTTCCBCCCCBSTTCCSCTTTCCHHHHTTBCCCHHHHHHHHHHHHHHHHHSSCTTSSCCCSSGGGHHHHHHHHHHHSCHHHHTTTSCHHHHHHHHHHHHSCSSCCCCHHHHHHHHHHHHHHTTCCCSCCCGGGCDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDDD <span class="at">-elastic</span> <span class="at">-ef</span> 700.0 <span class="at">-el</span> 0 <span class="at">-eu</span> 0.85 <span class="at">-eunit</span> 1:292</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>In the near future, the Martini3-IDP model will be fully implemented in Martinize2, simplifying the generation of multi-domain protein models with Martini 3.</p>
</section>
</section>
<section id="i.iii.2-simulation-of-the-multi-domain-ck1Œ¥-model" class="level2">
<h2 class="anchored" data-anchor-id="i.iii.2-simulation-of-the-multi-domain-ck1Œ¥-model">I.III.2 Simulation of the Multi-Domain CK1Œ¥ Model</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Fig10.png" class="img-fluid figure-img"></p>
<figcaption>Figure 10: Simulation of CK1Œ¥ including the IDR using the modified water bias (A) and the Martini3-IDP model (B). Displayed are 100 frames of a 5 Œºs simulation, with the structured region shown in gray and the IDR in pink.</figcaption>
</figure>
</div>
<p>As described in <a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut1.html">Section I.I.4</a>, the system can be solvated using insane with the following command.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">insane</span> <span class="at">-f</span> CK1d_cg.pdb <span class="at">-o</span> CG.gro <span class="at">-p</span> system.top <span class="at">-pbc</span> cubic <span class="at">-box</span> 35,35,35 <span class="at">-salt</span> 0.15 <span class="at">-sol</span> W <span class="at">-d</span> 0 </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>When simulating a protein with an IDR, it is especially important to use an appropriately sized box, to prevent the extended IDR from interacting with its mirror image. We therefore use an edge length of 35 nm here. The simulation of the system follows the protocol outlined in <a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut1.html">Section I.I.4</a>.</p>
<p>The conformational ensemble sampled during the simulations is shown in Figure 10.</p>
<hr>
<p>The following and final section (<a href="../../../../docs/tutorials/Martini3/ProteinsI/Tut4.html">I.IV</a>) will provide some additional information and discuss some limitations of the Martini 3 protein model, as well as provide a quick summary and outlook of this tutorial.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "Óßã";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/cgmartini\.nl");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<div class="footer-left">

</div>
</div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://x.com/cg_martini">
      <i class="bi bi-twitter" role="img" aria-label="Martini Twitter">
</i> 
    </a>
  </li>  
    <li class="nav-item">
    <a class="nav-link" href="../../../../docs/faq/index.html">
<p><iconify-icon inline="true" role="img" icon="hugeicons:help-circle" style="font-size: 1.25em;" aria-label="FAQs" title="FAQs"></iconify-icon></p>
</a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/Martini-Force-Field-Initiative">
      <i class="bi bi-github" role="img" aria-label="Martini GitHub">
</i> 
    </a>
  </li>  
</ul>
    </div>
    <div class="nav-footer-right">
<div class="footer-right">
¬© 2025 Martini Force Field Initiative <img src="https://cgmartini-library.s3.ca-central-1.amazonaws.com/icon.png" alt="Icon">
</div>
</div>
  </div>
</footer>
<script>
    document.addEventListener('DOMContentLoaded', (event) => {
      const quotes = [
          "<i>Shaken, not stirred.</i> - James Bond",
          "<i>He was white and shaken, like a dry martini.</i> - P. G. Wodehouse",      
          "<i>I think people think we're all sipping martinis by the pool.</i> - Maria Menounos",
          "<i>Why don't you get out of that wet coat and into a dry martini?</i> - Robert Benchley",
          "<i>Really, can anyone drink several martinis at lunch?</i> - Christine Baranski",
          "<i>Zen martini: A martini with no vermouth at all. And no gin, either.</i> - P.J. O'Rourke",
          "<i>I never go jogging, it makes me spill my martini.</i> - George Burns",
          "<i>Martinis are the only American invention as perfect as a sonnet.</i> - H. L. Mencken",
          "<i>He knows just how I like my martini: full of alcohol.</i> - Homer Simpson",
          "<i>Happiness is: finding olives in your martini when you're hungry.</i> - Johnny Carson",
          "<i>If it wasn't for the olives in his martinis he'd starve to death.</i> - Milton Berle",
        ];
      const randomIndex = Math.floor(Math.random() * quotes.length);
      const randomQuote = quotes[randomIndex];

      const footerLeft = document.querySelector('.footer-left');
      if (footerLeft) {
          footerLeft.innerHTML = randomQuote;
      }
    });
</script>




</body></html>